crates/common/types/transaction.rs

```diff
@@ -939,6 +940,172 @@ mod serde_impl {
         }
     }
 
+    impl<'de> Deserialize<'de> for Transaction {
```

Comment:
Why do we need to implement deserialize manually? Do we need to deserialize a special format?

Reply:
we need `de/serialization` for the prover.

Reply:
it has to be a custom Deserialization method because we have a custom Serialization method for all transactions

Reply:
We refactored this to derive Deserialize for Transaction

Reply:
Ok, let's add some comments stating why we're doing this, so future readers have context on why we're doing it this way, IMO this intent is not really clear from the code alone, but feel free to correct me.